stl/inc/xloctime

```diff
@@ -539,6 +539,10 @@ protected:
 private:
     ios_base::iostate __CLRCALL_OR_CDECL _Getint(_InIt& _First, _InIt& _Last, int _Lo, int _Hi, int& _Val,
         const _Ctype& _Ctype_fac) const { // get integer in range [_Lo, _Hi] from [_First, _Last)
+        const int _Hi_digits =
+            (_Hi <= 9 ? 1
+                      : _Hi <= 99 ? 2
+                                  : _Hi <= 999 ? 3 : _Hi <= 9999 ? 4 : static_cast<int>(_STD log10(_STD abs(_Hi))) + 1);

```

Comment:
This code is correct, but upon looking at it again, I noticed that `_STD abs(_Hi)` is unnecessary, because negative values would already have been absorbed by `_Hi <= 9 ? 1`. Then I wondered whether we need to worry about negative or enormous values at all - and I believe we don't. This is an internal function, and we pass constants for all of the `_Lo` and `_Hi` values. `_Hi` is always positive, and at most `9999`:
https://github.com/microsoft/STL/blob/5f736efd868f0cc32dd4c89d43f8ad9923742647/stl/inc/xloctime#L357
Therefore, I think it would be simpler (and less codegen in debug mode) to reduce the logic here, and add `_STL_INTERNAL_CHECK`s (which are assertions activated only for the STL's test suite, never for user compilation) to detect maintenance issues in the future.

Reply:
I didn't know about `_STL_INTERNAL_CHECK`. So, I tried to let `_Getint` be general and valid for possible future usages. Using such assertation is very reasonable here. :+1: